### ISSUE TITLE
Sync hangs with --track-renames and --track-renames-strategy "modtime" flags and correct deletions/transfers are not performed

### ISSUE THREAD

**Issue by Mushin** [rclone/rclone#4427]
#### What is the problem you are having with rclone?

When using the --track-renames and --track-renames-strategy "modtime" flags, the checker is skipping many files and is not performing the correct transfers and deletions. See below (note, the 1534 transfers and 2150 deletions are correct and should be performed):

without --track-renames --track-renames-strategy "modtime":

```
root@nas:~# rclone sync "/mnt/user/test" "crypt:current/test" --progress --fast-list --max-backlog 10000000 --transfers 3 --tpslimit 3 --checkers 16 --dry-run --log-file "/mnt/user/scripts/rclone/logs/testing.txt" -vv
Transferred:             0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:              4820 / 4820, 100%
Deleted:             2150
Transferred:         1534 / 1534, 100%
Elapsed time:         0.6s
```

with --track-renames --track-renames-strategy "modtime":

```
rclone sync "/mnt/user/test" "crypt:current/test" --progress --fast-list --max-backlog 10000000 --transfers 3 --tpslimit 3 --checkers 16 --log-file "/mnt/user/scripts/rclone/logs/testing.txt" -vv --dry-run --
track-renames --track-renames-strategy "modtime"
Transferred:             0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:              2684 / 2684, 100%
Transferred:           11 / 11, 100%
Elapsed time:         0.4s
```

Also, the sync process is hanging at this point - the last log line is:

`2020/07/11 12:22:40 DEBUG : Encrypted drive 'crypt:current/test': Waiting for renames to finish`

#### What is your rclone version (output from `rclone version`)

```
rclone v1.52.2-192-g8d5bc7f2-beta
- os/arch: linux/amd64
- go version: go1.14.4
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Unraid 64 bit / Docker

####  Which cloud storage system are you using? (eg Google Drive)

Google Drive crypt

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone sync "/mnt/user/test" "crypt:current/test" --progress --fast-list --max-backlog 10000000 --transfers 3 --tpslimit 3 --checkers 16 --track-renames --track-renames-strategy "modtime" --dry-run --log-file "/mnt/user/scripts/rclone/logs/testing.txt" -vv`

Note the same error still happens without all the other additional flags, i.e.:

`rclone sync "/mnt/user/test" "crypt:current/test" --dry-run --track-renames --track-renames-strategy "modtime"
`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

[log.txt](https://github.com/rclone/rclone/files/4906698/log.txt)

**Comment by ncw** [comment]
Can you kill -QUIT the rclone process when it is hanging and post the backtrace please. That should help shows what rclone is doing there. It might be some kind of deadlock.

I think that is the thing to work on first 

You'd expect fewer transfers when using track renames, rclone should rename files which don't need to be transferred.

**Comment by Mushin** [comment]
Log from kill -QUIT: [log2.txt](https://github.com/rclone/rclone/files/4907468/log2.txt)

I agree that in general there should be fewer transfers with renaming, but in this case the vast majority of changes were deletions and new files, not renames.

**Comment by ncw** [comment]
I recently posted a fix for this on [the rclone forum](https://forum.rclone.org/t/track-renames-strategy-modtime-doesnt-work/16992) can you give it a go as I'm pretty sure it is the same problem.

[v1.52.2-215-g81716d30-fix-track-renames-deadlock-beta](https://beta.rclone.org/branch/v1.52.2-215-g81716d30-fix-track-renames-deadlock-beta/) on branch [fix-track-renames-deadlock](https://github.com/rclone/rclone/tree/fix-track-renames-deadlock) (uploaded in 15-30 mins)

**Comment by Mushin** [comment]
Thanks. I'm using the Unraid Rclone Beta plugin, and I think it will only update when I restart the server? I will let you know when I'm able to do so.

**Comment by ncw** [comment]
I've had confirmation on the forum that this fix works, so I've merged it to the beta now :-)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.53 or v1.52.3 whichever comes first.
